pkg/appgw/requestroutingrules.go

```diff
@@ -110,7 +110,7 @@ func (c *appGwConfigBuilder) RequestRoutingRules(ingressList []*v1beta1.Ingress)
 			defaultBackendID := generateBackendID(ingress, nil, nil, defBackend)
 
 			defaultHTTPSettings := c.backendHTTPSettingsMap[defaultBackendID]
-			defaultAddressPool := c.backendPoolMap[defaultBackendID]
+			defaultAddressPool := c.NewBackendPoolMap()[defaultBackendID]
```

Comment:
Should we call `c.NewBackendPoolMap()` once before the `for ingress ...` loop begins?